framework-core/src/main/java/org/daisy/pipeline/datatypes/DatatypeService.java

```diff
@@ -0,0 +1,97 @@
+package org.daisy.pipeline.datatypes;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Map;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.Source;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.URIResolver;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.Document;
+import org.xml.sax.SAXException;
+
+public class DatatypeService {
```

Comment:
Would it be useful to have a more general validate(String s) method, in addition to asDocument()? I'm thinking about situations were an XML representation of the datatype is not possible.